store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -493,6 +527,9 @@ public CarbonWriterBuilder withCsvInput(Schema schema) {
    */
   public CarbonWriterBuilder withCsvInput(String jsonSchema) {
     Objects.requireNonNull(jsonSchema, "schema should not be null");
+    if (this.schema == null) {
```

Comment:
Isn't it should be `if (this.schema != null)`, because we need to check the schema is set only once

Reply:
resolved